WIP decookies in JSLigo

Gau-thier opened the pull request:
closes #884 

~WIP:~ 
~- Contract is not up-to-date, need to re-compile it via `ligo compile contract` but can't make the command working great because of the import...........~

d4hines commented on nix/deku-p/service.nix:
```diff
@@ -7,7 +7,7 @@
   cfg = config.services.deku-node;
   listToString = lib.strings.concatStringsSep ",";
-  environment = builtins.mapAttrs (_: value: builtins.toString value) cfg.environment;
+  cookieVM = "${pkgs.nodejs-16_x}/bin/node ${deku-packages.${config.nixpkgs.system}.decookies-vm}/lib/node_modules/decookies-vm/lib/src/index.js"
```
All the changes to this file should be discarded.

gsebil08 replied:
done

d4hines commented on flake.nix:
```diff
@@ -80,6 +80,7 @@
             deku = self'.packages.deku;
             tuna = self'.packages.tuna;
             deploy-rs = deploy-rs.packages.${system}.default;
+
```
```suggestion
```

d4hines commented:
Also please remove the _esy stuff.

d4hines commented:
Looks good but please clean the commit history, something like:
- one commit for the vm fixes
- one commit for the protocol fixes
- one comit for deku-c

gsebil08 commented:
>Looks good but please clean the commit history, something like:
one commit for the vm fixes
one commit for the protocol fixes
one comit for deku-c

Aren't we doing squash and merge on main? If yes, all these commits will be combined into only one... Hence, I don't understand/see why it is useful to rework all of them...

d4hines commented:
@gsebil08 we are doing squash and merge only when it makes sense - i.e when the PR is pretty much atomic already. This PR is several discrete changes. When we are debugging, we want to keep them separate.

Commit:
fix deku-c WASM VM

Commit:
fix deku protocol

Commit:
decookies in JSLigo

Commit:
only positive on nat?

Commit:
update contract and parameters

Commit:
uncomment decookie tests

Commit:
fix eat and transfer

d4hines committed:
feat: move ligo rpc into monorepo

d4hines committed:
refactor: move to a single library

d4hines committed:
refactor: rename main module

d4hines committed:
feat: add tuna support

Also does a major refactor of the ligo rpc server
to match how deku-api is implemented

d4hines committed:
feat: use new ligo rpc endpoints

d4hines committed:
fix: add tsconfig to deku-cli

d4hines committed:
feat: switch to prod ligo-deku-rpc

d4hines committed:
feat: include default url's in deku c client

d4hines committed:
fix: typo's in documentation

Commit:
contract for delegated keys

d4hines commented on deku-c/deku-cli/index.js:
```diff
@@ -0,0 +1,14 @@
+"use strict";
```
Looks like this file escaped .gitignore mistakenly.

gsebil08 replied:
Yup, done.

d4hines commented on deku-p/src/cli/mock_transaction.ml:
```diff
@@ -0,0 +1,74 @@
+(* We've dropped support for the External_vm currently.
```
This file should be deleted.

d4hines commented on deku-p/src/core/bin/node/deku_node.ml:
```diff
@@ -118,9 +118,9 @@ let main params style_renderer log_level =
-  (* The VM must be started before the node because this call is blocking
-     Logs.info (fun m ->
-         m "Starting IPC with external vm at path %s" named_pipe_path); *)
+  (* The VM must be started before the node because this call is blocking 
```
Be sure to run `dune build @fmt --auto-promote`.

gsebil08 replied:
Done.

d4hines commented on decookies/wasm-vm/src/types.jsligo:
```diff
@@ -32,15 +33,29 @@ type actions =
+    layerOneAddress: address
+  };
+
+type delegate = {layerOneAddress: address, address: address};
```
I don't think this operation should have a second `address` field. I think this should come directly from `Tezos.get_sender ()`.

Commit:
fix all the tests

Commit:
review

Commit:
review, bis

gsebil08 commented on decookies/wasm-vm/src/types.jsligo:
```diff
@@ -32,15 +33,29 @@ type actions =
+    operation: actions,
     amount: nat,
-    recipient: option<address>
+    layerOneAddress: address
```
You are right, naming was confusing. Changed to `onBehalfOf` everywhere.

Commit:
verify authorized keys

Commit:
format

Commit:
fix tests

Commit:
re-generate new contract

Commit:
re-generate every parameters

Commit:
fix tests

Commit:
fix decookies tests

Commit:
fix decookies tests

Commit:
test every buildings

Commit:
remove useless print

Commit:
test transfer feature

Commit:
switch to big_map
